### PR TITLE
[Hotfix] Fix Assessment Lock View and Null Index Column when locking Submission

### PR DESCRIPTION
**Pull request by fde312**
### Checklist
- [x] I tested *all* changes and *all* related features locally
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Wrong routing when clicking on `Open assessment` and nothing happened when clicking on `Cancel assessment`.
Null index issue when locking submission.

### Description

- Add logic for programming exercises
- Fixed null index issue when locking submission
- Removed not needed code 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercise, participate and go to assessment dashboard
4. Start new assessment of some submission but DO NOT submit
5. Navigate back
6. Click on `Assessments locked by you`
7. Check that you can open and cancel the assessment
8. Try this a few times and check if everything works as expected

### Test Coverage

Nothing changed

### Screenshots

video

### CODE REVIEW

**Review by airelawaleria — COMMENTED** [review]
We noticed with @simonzilker  that if you open an assessment from "Assessments locked by you", and then go back, you are no more in "assessments locked by you", but at the "exercise dashboard" page. Is it intended? Because on the screenshots it looks otherwise

**Review by simonzilker — COMMENTED** [review]
tested with @airelawaleria  that when you open an assessment from "Assessments locked by you", and then go back on the "Artemis back button", you are no more in "Assessments locked by you". It redirects to the "exercise dashboard" page. But when you click the back button in the browser, you are redirected to " assessments locked by you".Is it intended? Because on the screenshots it looks different.

**Comment by fde312** [comment]
> tested with @airelawaleria that when you open an assessment from "Assessments locked by you", and then go back on the "Artemis back button", you are no more in "Assessments locked by you". It redirects to the "exercise dashboard" page. But when you click the back button in the browser, you are redirected to " assessments locked by you".Is it intended? Because on the screenshots it looks different.

@airelawaleria @simonzilker Thank you for testing! This isan issue with the navigate back button and not related to this PR. As we want to get rid of the back button in the next future and just use the breadcrumb navigation I would not consider this as a stopper to merge this PR :) If the rest worked as expected, please approve :D

**Review by airelawaleria — APPROVED** [review]
Works as intended!

**Review by simonzilker — APPROVED** [review]
Everything worked as stated :)

**Review by anditurdiu — APPROVED** [review]
Code looks good

**Review by JohannesWeiss — APPROVED** [review]
Works as intended on ts1

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts [hunk 1]
```diff
@@ -113,17 +113,19 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
...
+                } else if (response?.error?.status === 404) {
                     // there are no unassessed submission, nothing we have to worry about
                     this.onError('artemisApp.exerciseAssessmentDashboard.noSubmissions');
-                } else if (this.route.snapshot.data.studentParticipationId.error) {
-                    this.onError(this.route.snapshot.data.studentParticipationId.error);
+                    return;
+                } else if (response?.error) {
+                    this.onError(response?.error);
                     return;
                 }
```
Comment (jpbernius) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts:127]: Nice use of optional chaining!

**Review by jpbernius — APPROVED** [review]

**Pull request merged**